src/us/kbase/workspace/test/kbase/HandleAndBytestreamIntegrationTest.java

```diff
@@ -556,14 +553,106 @@ public void basicHandleTest() throws Exception {
+						is("The Handle Service reported a problem while attempting to set Handle ACLs: 'Unable to set acl(s) on handles " + handle_id + "'"));
 		assertThat("incorrect stacktrace", wod.getHandleStacktrace(),
 						startsWith("us.kbase.typedobj.idref.IdReferencePermissionHandlerSet$" +
 										"IdReferencePermissionHandlerException: " +
-										"The Handle Manager reported a problem while attempting to set Handle " +
+										"The Handle Service reported a problem while attempting to set Handle " +
 										"ACLs: 'Unable to set acl(s) on handles " + handle_id));
 	}
 
+	public String createHandle(final String blobID, final String url, final String fileName)
+			throws Exception {
+		return HANDLE_CLIENT.persistHandle(new Handle()
+				.withId(blobID)
+				.withFileName(fileName)
+				.withUrl(url)
+				.withType("shock"));
+	}
+
+	@Test
+	public void handlesWithBatchUpdates() throws Exception {
+		/* Tests the batch updates parameter. Note that this has no visible impact to the user
+		 * other than a possible performance improvement, which is not reasonable to test for
+		 * as the performance characteristics can vary wildly from test environment to test
+		 * environment.
+		 * As such, all this test really does is check that the blobstore nodes are updated as
+		 * expected and no errors occur.
+		 * I did manually check that the handle ID handler was getting the batched IDs vs.
+		 * single object IDs.
+		 */
+		
+		final String workspace = "batchhandle";
+		CLIENT1.createWorkspace(new CreateWorkspaceParams().withWorkspace(workspace));
+		final BasicShockClient bsc = new BasicShockClient(
+				new URL("http://localhost:" + BLOB.getPort()), CLIENT1.getToken());
+
+		final ShockNode node1 = addBasicNode(bsc);
+		final ShockNode node2 = addBasicNode(bsc);
+		final String blob_id1 = node1.getId().getId();
+		final String blob_id2 = node2.getId().getId();
+		final String hid1 = createHandle(blob_id1, bsc.getShockUrl().toString(), "foo");
+		final String hid2 = createHandle(blob_id2, bsc.getShockUrl().toString(), "bar");
+
+		try {
+			CLIENT1.saveObjects(new SaveObjectsParams()
+					.withWorkspace(workspace)
+					.withObjects(Arrays.asList(
+							new ObjectSaveData()
+									.withData(new UObject(ImmutableMap.of(
+											"handles", Arrays.asList(hid1))))
+									.withName("foo")
+									.withType(HANDLE_TYPE),
+							new ObjectSaveData()
+									.withData(new UObject(ImmutableMap.of(
+											"handles", Arrays.asList(hid2))))
+									.withName("bar")
+									.withType(HANDLE_TYPE)
+							))
+					);
+		} catch (ServerException se) {
+			System.out.println(se.getData());
+			throw se;
+		}
+		
+		CLIENT1.setPermissions(new SetPermissionsParams().withWorkspace(workspace)
+				.withUsers(Arrays.asList(USER2)).withNewPermission("r"));
+		final GetObjects2Params params = new GetObjects2Params()
+				// comment this out to check batching manually, need to add use debugger or
```

Comment:
`add` has been removed locally, it'll be fixed whenever I get around to making another WS PR